sa/sa.go

```diff
@@ -1075,6 +1075,54 @@ func addFQDNSet(tx *gorp.Transaction, names []string, serial string, issued time
 	})
 }
 
+// addOrderFQDNSet creates a new OrderFQDNSet row using the provided
+// information. This function doesn't accept a transaction because it must be
+// performed after the add transaction has completed in order for the orderID to
+// be known.
+func addOrderFQDNSet(
+	dbMap *gorp.DbMap,
+	names []string,
+	orderID int64,
+	regID int64,
+	expires time.Time) error {
+	return dbMap.Insert(&core.OrderFQDNSet{
+		SetHash:        hashNames(names),
+		OrderID:        orderID,
+		RegistrationID: regID,
+		Expires:        expires,
+	})
+}
+
+// deleteOrderFQDNSet deletes a OrderFQDNSet row that matches the provided
+// information. This function accepts a transaction so that the deletion can
+// take place within the finalization transaction.
+func deleteOrderFQDNSet(
+	tx *gorp.Transaction,
+	names []string,
+	orderID int64,
+	regID int64) error {
+
+	result, err := tx.Exec(`
+	  DELETE FROM orderFqdnSets
+		WHERE setHash = ?
+		AND orderID = ?
+		AND registrationID = ?`,
+		hashNames(names),
+		orderID,
+		regID)
+	if err != nil {
+		return Rollback(tx, err)
+	}
+	// We always expect there to be an order FQDN set row for each
+	// pending/processing order that is being finalized. If there isn't one then
+	// something is amiss and should be raised as an internal server error
+	if rowsDeleted, err := result.RowsAffected(); err != nil || rowsDeleted == 0 {
```

Comment:
Just in case, let's split out the err != nil case (where we'd want to log the error) from the rowsDeleted == 0 error.

Reply:
:+1: - Fixed in 3a68e54